app.js

```diff
@@ -232,48 +307,31 @@ var hotelA = new Hotel('thisisahotel','itliveshere','hotelPlaceholder.jpg','hote
 window.addEventListener('load', onLoad);
 
 window.addEventListener('click', function(event){
+  if(event.target.id === 'button'){
+    return;
+  }
   hotelA.displayRoom(event);
 });
 
-var submitClick = document.getElementById('submit');
-submitClick.addEventListener('click', function(event){
+var dropdown = document.getElementById('dropdown-box');
+dropdown.addEventListener('change', function(){
   filterRooms();
+  checkBoxFilter(roomsAvailable);
 });
 
-var roomsAvailable = [];
-var roomsNotAvailable = [];
-
-function filterRooms(){
-  event.preventDefault();
-  var dropdownBox = document.getElementById('dropdown-box');
-  var roomTypeSelected = dropdownBox.options[dropdownBox.selectedIndex].value;
-  roomsAvailable = [];
-  roomsNotAvailable = [];
-  for (var key in hotelA.hotelRooms){
-    if (roomTypeSelected === hotelA.hotelRooms[key].roomType && hotelA.hotelRooms[key].isVacant){
-      roomsAvailable.push(key);
-      var availableRooms = document.getElementById(key);
-      availableRooms.style.fill = '#1ea83c';
-    }
-    if (roomTypeSelected !== hotelA.hotelRooms[key].roomType) {
-      roomsNotAvailable.push(key);
-      var unavailableRooms = document.getElementById(key);
-      unavailableRooms.style.fill = '#919191';
+var checkboxes = (document.querySelectorAll('.checkBox'));
+checkboxes.forEach(function(item){
+  item.addEventListener('change', function(event){
+    if (document.getElementById(event.target.id).checked) {
```

Comment:
`document.getElementById(event.target.id)` should be the same as `event.target`. Did you find that not to be the case?